node/consensus/src/lib.rs

```diff
@@ -212,6 +212,11 @@ impl<N: Network> Consensus<N> {
 impl<N: Network> Consensus<N> {
     /// Adds the given unconfirmed solution to the memory pool.
     pub async fn add_unconfirmed_solution(&self, solution: ProverSolution<N>) -> Result<()> {
+        #[cfg(feature = "metrics")]
```

Comment:
I think we should probably add this to after the unconfirmed solutions and transmissions are added, since there are a few points where we fail or return early

Reply:
Initially I did that, but @vicsn requested to move to top.

Reply:
yep, I saw that. I think if we want it at the top we'd have to add metrics flagged code to decrement the gauge on failure or early return, which probably isn't optimal

Reply:
I think the point is to see the difference with how many come in and how many actually make it into a block.

Reply:
Fair, although in that case it would probably be better to track it from the delivering side, i.e. whatever we are sending. In this case, we are omitting metrics maybe one layer too deep in the code since what we really want is to omit metrics whenever these post endpoints [are called](https://github.com/AleoHQ/snarkOS/blob/mainnet/node/rest/src/routes.rs#L312). If we do want to just track how many times those endpoints are hit it might be cleaner to move it there- @vicsn what do you think?

Reply:
Rationale for putting it in `add_unconfirmed_{transaction, solution}` is so we also count transactions received via the router.

I think measuring it on the delivering side (i.e. tx-cannon) would be even better, but that requires more coordination and time to figure out.